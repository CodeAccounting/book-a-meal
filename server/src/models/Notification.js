import Sequelize from 'sequelize';

export default (sequelize) => {
  const Notification = sequelize.define(
    'Notification',
    {
      notifId: {
        type: Sequelize.UUID,
        primaryKey: true,
        defaultValue: Sequelize.UUIDV4,
        allowNull: false
      },
      message: {
        type: Sequelize.TEXT,
        allowNull: false
      },
    },
  );

  Notification.associate = (models) => {
    Notification.belongsTo(models.User, {
      foreignKey: 'userId',
      onDelete: 'CASCADE',
    });

    Notification.belongsTo(models.Menu, {
      foreignKey: 'menuId',
      onDelete: 'CASCADE',
    });

    Notification.belongsTo(models.Order, {
      foreignKey: 'orderId',
      onDelete: 'CASCADE',
    });
  };

  return Notification;
};
