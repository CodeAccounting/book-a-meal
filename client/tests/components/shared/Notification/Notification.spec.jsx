import React from 'react';
import Notification from '../../../../src/components/shared/Notification';

describe('Notification', () => {
  it('renders correctly', () => {
    const shallowWrapper = shallow(<Notification message="Hi, you rock" />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});
