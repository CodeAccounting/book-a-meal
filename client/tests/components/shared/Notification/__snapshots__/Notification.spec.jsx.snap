// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders correctly 1`] = `
<div
  className="notification-pill"
>
  <p
    className="text-center notif"
  >
    Hi, you rock
  </p>
</div>
`;
